include/nbl/ui/CCursorControlXCB.h

```diff
@@ -0,0 +1,34 @@
+#ifndef __NBL_SYSTEM_C_CURSOR_CONTROL_XCB_H_INCLUDED__
+#define __NBL_SYSTEM_C_CURSOR_CONTROL_XCB_H_INCLUDED__
+
+#ifdef _NBL_PLATFORM_LINUX_
+
+#include "nbl/ui/ICursorControl.h"
+
+namespace nbl::ui
+{
+
+class XCBConnection;
+class NBL_API2 CCursorControlXCB final : public ICursorControl
+{
+    public:
+        inline CCursorControlXCB(
+        core::smart_refctd_ptr<XCBConnection>&& xcbConnection) : 
+            m_connection(std::move(xcbConnection)) {}
+
+        void setVisible(bool visible) override;
+        bool isVisible() const override;
+
+        void setPosition(SPosition pos) override;
+        void setRelativePosition(IWindow* window, SRelativePosition pos) override;
+
+        SPosition getPosition() override;
+        SRelativePosition getRelativePosition(IWindow* window) override;
+    private: 
+        core::smart_refctd_ptr<XCBConnection> m_connection;
+};
+}
```

Comment:
I dunno if its a good idea in your case, but for the Win32 stuff (#476) it ended up making more sense to have the manager inherit both from `ICursorControl` and `IWindowManager` because of the global nature of the cursor on Windows.

Don't know if this is applicable here

Reply:
not sure how that works in terms of xcb I know x11 supports more then one cursor but applications tends to catch fire if you try to set it up like that ...